src/lib/geometry.f90

```diff
@@ -2588,6 +3145,53 @@ function coord_at_xi(ne,xi,basis)
 
   end function coord_at_xi
 
+
+!
+!###################################################################################
+!
+  subroutine splitwords(line, words, nw )
+    implicit none
+    character(*), intent(in)  :: line
+    character(*), intent(out) :: words(:)
+    integer,      intent(out) :: nw
+    character(len(words)) :: buf( size(words) )
+    integer :: k, ios
+
+    nw = 0 ; words(:) = ""
+
+    do k = 1, size(words)
+        read( line, *, iostat=ios ) buf( 1 : k )
+        if ( ios /= 0 ) exit
+        nw = k
+        words( 1 : nw ) = buf( 1 : nw )
+    enddo
+
+  end subroutine splitwords
+
+
+!###################################################################################
+
+  subroutine words_to_real( words,ints, ni )
```

Comment:
Subroutine or function? Belongs in Geometry or utilities?

Reply:
@tawhai please also comment on this philosophical point. In the geometry module we have a number of functions taht are doing things that essentially check numbers that are input. etc. in general these are functions not subroutines, but perhaps a greater question is do these belong in the geometry module at all - they have nothing to do really with geometry, and are utilities associated with this codebase, as far as I interpret them.

Reply:
@tawhai i think i worded my comment a little vaguely here. I wasn't asking for you to make any changes, this is @haribalankumar's pull request, seperate from your recent one. Just wanted you to take a look at these new subroutines he has added, and comment on whether you think they are subroutines or functions, and whether this type of function (from a lungsim philosophy perspective) belongs in the geometry module? Sorry for not being clearer!